db/external_sst_file_basic_test.cc

```diff
@@ -351,6 +353,19 @@ TEST_F(ExternalSSTFileBasicTest, BasicWithFileChecksumCrc32c) {
   DestroyAndRecreateExternalSSTFilesDir();
 }
 
+TEST_F(ExternalSSTFileBasicTest, KeyValueOrdering) {
+  Options options = CurrentOptions();
```

Comment:
I didn't see paranoid file checks set here, so I'm not sure the new code path is covered.

Reply:
Sure. But even the case where it goes through the code and makes sure it doesn't break or report corruption while the data is OK, is valuable to be included.
Regarding corrupt a file, db/corruption_test.cc should have exanples.

Reply:
By corrupting a file, wouldn't the read-back encounter block checksum errors?

I'd try corrupting the key and/or value in-memory around `builder->Add()`. You can set a sync-point like https://github.com/facebook/rocksdb/blob/b6cfda12837006c535c85f5256ae4a6e2878b75a/db/memtable.cc#L569 that passes the key/value data to the callback, then register a callback that corrupts the key/value using `SetCallBack()`.

Reply:
@siying @ajkr thanks for the feedback and tips. I just sent out a new commit that addresses your concerns. See if it makes sense